plugin-build/src/test/kotlin/io/sentry/android/gradle/tasks/dependencies/SentryExternalDependenciesReportTaskTest.kt

```diff
@@ -19,13 +19,27 @@ class SentryExternalDependenciesReportTaskTest {
     val project = createRegularProject()
     val output = tempDir.newFolder("dependencies")
 
-    val task: TaskProvider<SentryExternalDependenciesReportTask> =
+    val task: TaskProvider<SentryExternalDependenciesReportTaskV2> =
```

Comment:
This might be the change I am the least sure of.

I couldn’t find a test for this `SentryExternalDependenciesReportTaskV2` and it seems that the output of `V2` is different than the regular `SentryExternalDependenciesReportTask` because the task doesn’t have any information about the version numbers. Otherwise the logic is the same.

Reply:
Pre-stamped, but would be good to ensure this doesn't affect anything before merge. cc @romtsn

Reply:
Seems like it was only a workaround for making [our plugin more configuration cache friendly](https://github.com/getsentry/sentry-android-gradle-plugin/blob/8818afdfc6a705a8ab7df50dd8c988d5692239c7/plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/dependencies/SentryExternalDependenciesReportTaskFactory.kt#L26-L27). If the V2 implementation passes the V1 test, we should be good to go

Reply:
Hmm maybe we should replace `it.id.displayName` [here](https://github.com/getsentry/sentry-android-gradle-plugin/blob/8818afdfc6a705a8ab7df50dd8c988d5692239c7/plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/dependencies/SentryExternalDependenciesReportTaskV2.kt#L70C31-L70C38) with something like this:
```Kotlin
.map {
    val component = it.id as ModuleComponentIdentifier
    "${component.group}:${component.module}:${component.version}"
}
```

Reply:
hmm, something seems off. I just executed the task on the `main` branch and here's the scan: https://scans.gradle.com/s/en46g26szhnc4/timeline?details=dfdg4vqyhz6a2. You can see it uses the V2 task and if you switch over to console it outputs the deps with their respective versions.

Does it have to do with how we set up the test (i.e. not providing the versions in `artifactIds`)?